From 5a5a910c82633484b95c800a892811dbf601b70e Mon Sep 17 00:00:00 2001
From: Tomas Popela <tomas.popela@gmail.com>
Date: Thu, 26 Oct 2017 20:25:51 +0000
Subject: To M63: Cannot compile media/filters/source_buffer_stream.cc with GCC

Make the BufferingByPts() static to be able to compile it with GCC 6.3.
Clang compiles fine as well as GCC 7.2+.

Upstream GCC bug - https://gcc.gnu.org/bugzilla/show_bug.cgi?id=66297

TBR=dalecurtis@chromium.org

Bug: 777391
Cq-Include-Trybots: master.tryserver.chromium.android:android_optional_gpu_tests_rel;master.tryserver.chromium.linux:linux_optional_gpu_tests_rel;master.tryserver.chromium.mac:mac_optional_gpu_tests_rel;master.tryserver.chromium.win:win_optional_gpu_tests_rel
Change-Id: I9a3d0b1ad9abaff245f6a7ac540591acd990f106
Reviewed-on: https://chromium-review.googlesource.com/733121
Reviewed-by: Dale Curtis <dalecurtis@chromium.org>
Reviewed-by: Matthew Wolenetz <wolenetz@chromium.org>
Commit-Queue: Matthew Wolenetz <wolenetz@chromium.org>
Cr-Original-Commit-Position: refs/heads/master@{#511352}(cherry picked from commit 2083a5d15cbbe84b87e02075e397e6b1dcf1f4ca)
Reviewed-on: https://chromium-review.googlesource.com/740321
Cr-Commit-Position: refs/branch-heads/3239@{#249}
Cr-Branched-From: adb61db19020ed8ecee5e91b1a0ea4c924ae2988-refs/heads/master@{#508578}

diff --git a/media/filters/source_buffer_stream.h b/media/filters/source_buffer_stream.h
index 2ff5ba42c6af..93f8c0c4c472 100644
--- a/media/filters/source_buffer_stream.h
+++ b/media/filters/source_buffer_stream.h
@@ -397,7 +397,7 @@ class MEDIA_EXPORT SourceBufferStream {
   // TODO(wolenetz): Consider refactoring to reference a "buffering timestamp"
   // type (DTS for ByDts, PTS for ByPts) defined in RangeClass to reduce the
   // need for some of these helpers. See https://crbug.com/718641.
-  constexpr bool BufferingByPts();
+  static constexpr bool BufferingByPts();
   DecodeTimestamp BufferGetTimestamp(
       const scoped_refptr<StreamParserBuffer>& buffer);
   void RangeAppendBuffersToEnd(RangeClass* range,
-- 
2.11.0

